driver-core/src/main/com/mongodb/internal/authentication/GcpCredentialHelper.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.authentication;
+
+import com.mongodb.MongoClientException;
+import org.bson.BsonDocument;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mongodb.internal.authentication.HttpHelper.getHttpContents;
+
+/**
+ * Utility class for working with GCP authentication.
+ *
+ * <p>This class should not be considered a part of the public API.</p>
+ */
+public final class GcpCredentialHelper {
+    public static BsonDocument obtainFromEnvironment() {
+        String alternateHost = System.getenv("GCE_METADATA_HOST");
```

Comment:
I realize there is no test for this.  @kevinAlbs what do you think?

Reply:
I ended up not using a mock server, but if someone wants to use it they can always modify /etc/hosts so that the env var is not needed, right?

Reply:
Yes. Good point. Submitted https://github.com/mongodb/specifications/pull/1290 to remove the requirement.

Reply:
Removed the env var